商品購入機能

zinkoko-lab opened the pull request:
# What
商品購入機能
# Why
 商品購入可能にするため、商品購入機能を実装しました。
コードレビューお願いいたします。

# Gyazo
### 必要な情報を適切に入力して「購入」ボタンを押すと、商品の購入ができる動画
[![Image from Gyazo](https://i.gyazo.com/4698b2bde5216ca4c99a181838aaea54.gif)](https://gyazo.com/4698b2bde5216ca4c99a181838aaea54)
### 入力に問題がある状態で「購入」ボタンが押された場合、情報は受け入れられず、購入ページでエラーメッセージが表示される動画
[![Image from Gyazo](https://i.gyazo.com/2e7517d33fb0971dfe3a6077d165b46f.gif)](https://gyazo.com/2e7517d33fb0971dfe3a6077d165b46f)
### ログイン状態の場合でも、URLを直接入力して自身が出品していない売却済み商品の商品購入ページへ遷移しようとすると、トップページに遷移する動画
[![Image from Gyazo](https://i.gyazo.com/87745f12ecc5a788562a7451b41767dc.gif)](https://gyazo.com/87745f12ecc5a788562a7451b41767dc)
### ログイン状態の場合でも、URLを直接入力して自身が出品した商品の商品購入ページに遷移しようとすると、商品の販売状況に関わらずトップページに遷移する動画
[![Image from Gyazo](https://i.gyazo.com/c4760ec6cc910494897a724b7eb09c74.gif)](https://gyazo.com/c4760ec6cc910494897a724b7eb09c74)
### ログアウト状態の場合は、URLを直接入力して商品購入ページに遷移しようとすると、商品の販売状況に関わらずログインページに遷移する動画
[![Image from Gyazo](https://i.gyazo.com/86d5a61d13aca126c5472ef4cf945653.gif)](https://gyazo.com/86d5a61d13aca126c5472ef4cf945653)
### 売却済みの商品は、画像上に「sold out」の文字が表示される動画（商品一覧機能実装時に未実装であった場合）
[![Image from Gyazo](https://i.gyazo.com/3697f33677d6aba6da0dcfa6d52eb0c8.gif)](https://gyazo.com/3697f33677d6aba6da0dcfa6d52eb0c8)
###  売却済みの商品は、画像上に「sold out」の文字が表示される動画（商品詳細機能実装時に未実装であった場合）
[![Image from Gyazo](https://i.gyazo.com/03e3920a417e12f76e1d11b8dbc69afe.gif)](https://gyazo.com/03e3920a417e12f76e1d11b8dbc69afe)
###  ログイン状態の場合でも、売却済みの商品には、「商品の編集」「削除」「購入画面に進む」ボタンが表示されない動画（商品詳細機能実装時に未実装であった場合）
[![Image from Gyazo](https://i.gyazo.com/0b5e3372336ee814f9f318879e2a3d25.gif)](https://gyazo.com/0b5e3372336ee814f9f318879e2a3d25)
###  ログイン状態の場合でも、URLを直接入力して自身が出品した売却済み商品の商品情報編集ページへ遷移しようとすると、トップページに遷移する動画（商品情報編集機能実装時に未実装であった場合）
[![Image from Gyazo](https://i.gyazo.com/c291226c8452417b4c5c143878f24df6.gif)](https://gyazo.com/c291226c8452417b4c5c143878f24df6)
### テスト結果の画像
[![Image from Gyazo](https://i.gyazo.com/e7646408903e24729d4a50ccc7fde191.png)](https://gyazo.com/e7646408903e24729d4a50ccc7fde191)

zinkoko-lab committed:
created_form_object

zinkoko-lab committed:
Add sold out functionality to items and orders; update application.js and importmap.rb

* In app/controllers/items\_controller.rb, added before\_action :sold\_out? to prevent editing, updating, and

zinkoko-lab committed:
Add Payjp payment functionality to OrdersController, card.js, and application.html.erb for purchasing items; update OrderShippingInfo model and OrdersController for token and price; include gon and Payjp library in

zinkoko-lab committed:
"Update order shipping info model, adjust test environment and remove unnecessary spec files"

This commit includes the following changes:

1. In the `OrderShippingInfo` model, the validation messages for the `postal_code`

koyamatakumi reviewed (commented):
jin5211様、レビュー依頼フォームの送信ありがとうございます。
コードを確認させていただき、コメントいたしましたのでご確認をお願いいたします。
修正後はコミット、プッシュを忘れずに行い、改めてレビュー依頼フォームの送信をお願い致します！

zinkoko-lab committed:
Add user has many associations with items and orders

In user model, add associations for has\_many :items and has\_many :orders.

In items view, display Sold Out message if item has an order.

In order

jin5211 commented:
# What
コード修正

# Why
コード修正しましたので、再度コードレビューをお願いいたします。

# Gyazo
[![Image from Gyazo](https://i.gyazo.com/9a9dcf2f3376e19aff6d2af5c5cd880c.png)](https://gyazo.com/9a9dcf2f3376e19aff6d2af5c5cd880c)

TAKADA-yuma reviewed (commented):
レビュー依頼ありがとうございます。プルリクエスト確認させていただきました。

修正点を追加でコメントさせて頂きましたので、ご確認をお願い致します。

```ruby
【次回プルリクエスト提出する際の注意点】
①修正が終わりましたら、`コミットとプッシュ`をして、`プルリクエストに修正内容が反映されているか確認`しましょう。
②既に頂いている挙動確認の画像や動画が修正範囲だった場合、最新の画像や動画の提出をお願い致します。
③googleフォームの送信をお願いいたします。
```


ご不明点や疑問点ございましたら、チャット・コネクトメンターにご相談ください！
どうぞよろしくおねがいします。

zinkoko-lab committed:
"Add spec for OrderShippingInfo model validations including token presence"

This commit adds a new spec to the OrderShippingInfo model that tests for the presence validation of the `token` attribute.

jin5211 commented:
# What
コード修正
# Why
コード修正しましたので、再度のコードレビューをお願いいたします。
# Gyazo
[![Image from Gyazo](https://i.gyazo.com/6df5cdaad94465e810df5376a89660b7.png)](https://gyazo.com/6df5cdaad94465e810df5376a89660b7)

koyamatakumi reviewed (commented):
jin5211様 レビュー依頼フォームの送信ありがとうございます！
**LGTM**とさせていただきます。実装お疲れさまでした！
マージ後、ローカルのmain(master)ブランチに忘れずにプルしましょう。
### ■ マージ後の手順
```ruby
 ローカルのCurrentBrunchをmain(master)に変更　→　FetchOrigin　→　PullOrigin
```
その後、次の作業に取り組んでいただけますと幸いです。
 休憩もはさみつつ引き続き頑張っていきましょう！
